File: Programmers/Stack,Queue/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
#include <queue>
#define TRUE 1
#define FALSE 0
using namespace std;


vector<int> solution(vector<int> progresses, vector<int> speeds)
{
    int temp, value;
    vector<int> answer;
    queue <int> s;
    for (int i = 0; i < progresses.size(); i++)
    {
        temp = 100 - progresses[i];
        if (temp % speeds[i] == 0)
            value = temp / speeds[i];
        else
        {
            value = (temp / speeds[i]) + 1;
        }
        if (s.empty())
        {
            s.push(value);
        }
        else
        {
            if (s.front() >= value)
                s.push(value);
            else
            {
                answer.push_back(s.size());
                while (!s.empty())
                    s.pop();
                s.push(value);
            }
        }
    }
    if (!s.empty())
        answer.push_back(s.size());
    return answer;
}

File: C&Python/Etc/기능개발.py
def solution(progresses, speeds):
    #변수 초기화
    idx = 0
    answer = []
    day = []
    
    # 배포까지 날짜 계산하기
    for i in range(len(progresses)):
        d = int((100 - progresses[i]) / speeds[i])
        r = int((100 - progresses[i]) % speeds[i])
        day.append(d)
        if r != 0 :
            day[i] += 1
    
    # 배포마다 answer 변수에 count
    answer.append(1)
    temp = day[0]
    for i in range(len(day) - 1):
        if(temp >= day[i + 1]):
            answer[idx] += 1
        else:
            idx += 1
            temp = day[i + 1]
            answer.append(1)    
    
    return answer